.github/workflows/add_assignee_when_pr_opened.yml

```diff
@@ -0,0 +1,48 @@
+# See https://confluence.diamond.ac.uk/x/tJqQC for instructions if this needs a new token
+name: Assign author of pull request to referenced issue that was not assigned to issue ticket beforehand
+on:   
+ pull_request:
+      types: [ready_for_review, opened, review_requested]
+jobs:
+  assign_pr_author_as_assignee:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get project data
+        # This will get information about the project in general and add it to the environment.
+        # * ISSUE_REFERENCED_ID - The ID for the Issue that was referenes at Fixes#999
+        # * AUTHOR_ID - The ID of the person who made a pull request so that the author can be assigned to the Fixes#999
+        env:
+          GITHUB_TOKEN: ${{ secrets.GHPROJECT_TOKEN }}
+          REPO_ORG_NAME: ${{ github.repository }}
+          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          gh api graphql -f query='
+            query ($org: String!, $repo: String!, $number: Int!) {
+              repository(owner: $org, name: $repo) {
+                pullRequest(number: $number) {
+                  closingIssuesReferences(first: 1) {
+                    edges {
+                      node {
+                        id
+                      }
+                    }
+                  }
+                  author { ... on User {id} }
+                }
+              }
+            }' -f org=$ORGANIZATION -f repo=${REPO_ORG_NAME##*/} -F number=$PULL_REQUEST_NUMBER > project_data.json
```

Comment:
Should: This is basically the same as `get_issue_from_pr.yml`. Could we just get that to return the author too and reuse the same job?